include/go_types.h

```diff
@@ -44,6 +44,13 @@ struct go_iface
     void *data;
 };
 
+struct map_bucket {
+    char tophash[8];
+    struct go_string keys[8];
+    struct go_slice values[8];
+    void *overflow;
```

Comment:
Is it used somewhere?

Reply:
You mean the whole map_bucket struct or specifically the overflow field?

The map_bucket struct is used by the context propagation code, and even if the overflow field is not directly used by our code it is still important to keep it as part of the map_bucket struct as we are iterating through the buckets of the map, and if remove this field we will end up reading wrong values from memory.

Reply:
I meant `overflow` field. Could you clarify why 

> we will end up reading wrong values from memory.

?

Reply:
Sure, I used the following resource to understand the Go map implementation:

https://github.com/cch123/golang-notes/blob/master/map.md

As you can see the map bucket struct has an overflow field, if we omit it from the map_bucket struct line 110 in the probe.bpf.c file which attempts to read the next bucket will not read it correctly as the loop advances.